pkg/kf/commands/install/gke.go

```diff
@@ -0,0 +1,908 @@
+	_, err = gcloud(
+		ctx,
+		"-q",
+		"services",
+		"--project", projID,
+		"enable",
+		serviceName,
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createServiceAccount(
+	ctx context.Context,
+	serviceAccountName string,
+	serviceAccount string,
+	projID string,
+) error {
+	ctx = setLogPrefix(ctx, "Create Service Account")
+	Logf(ctx, "Creating service account %s", serviceAccount)
+	if _, err := gcloud(
+		ctx,
+		"iam",
+		"service-accounts",
+		"create",
+		serviceAccountName,
+		"--project", projID,
+	); err != nil {
+		return err
+	}
+	if _, err := gcloud(
+		ctx,
+		"projects",
+		"add-iam-policy-binding",
+		projID,
+		"--member", "serviceAccount:"+serviceAccount,
+		"--role", "roles/storage.admin",
+	); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func ensureBilling(ctx context.Context, projID string) error {
+	ctx = setLogPrefix(ctx, "Ensure Billing")
+
+	enabled, err := billingEnabled(ctx, projID)
+	if err != nil {
+		return err
+	}
+	if enabled {
+		// Looks good, move on
+		return nil
+	}
+
+	Logf(ctx, "Looks like you need to enable billing for %s", projID)
+	ok, err := selectYesNo(ctx, fmt.Sprintf("Sync billing account for %s?", projID))
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return fmt.Errorf("no billing setup for %s", projID)
+	}
+
+	availableAccounts, err := billingAccounts(ctx)
+	if err != nil {
+		return err
+	}
+	_, accountID, err := selectPrompt(ctx, "Billing Account", availableAccounts...)
+	if err != nil {
+		return err
+	}
+
+	if err := linkBilling(ctx, projID, accountID); err != nil {
+		return err
+	}
+	return nil
+}
+
+// billingEnabled checks to see if a project has billing enabled
+func billingEnabled(ctx context.Context, projID string) (bool, error) {
+	Logf(ctx, "checking if %s has billing enabled", projID)
+	result, err := gcloud(
+		ctx,
+		"alpha",
+		"billing",
+		"projects",
+		"describe",
+		projID,
+		"--format",
+		"value(billingEnabled)",
+	)
+	if err != nil {
+		return false, err
+	}
+
+	return strings.ToLower(strings.Join(result, "")) == "true", nil
+}
+
+// billingAccounts will look to see what accounts the user has to help with
+// linking them.
+func billingAccounts(ctx context.Context) ([]string, error) {
+	Logf(ctx, "fetching billing accounts")
+	accounts, err := gcloud(
+		ctx,
+		"alpha",
+		"billing",
+		"accounts",
+		"list",
+		"--format",
+		"value(name)",
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(accounts) == 0 {
+		return nil, errors.New("there has to be at least one billing account setup")
+	}
+
+	return accounts, nil
+}
+
+// linkBilling links project to billing account.
+func linkBilling(ctx context.Context, projID, accountID string) error {
+	Logf(ctx, "linking %s to %s", projID, accountID)
+	_, err := gcloud(
+		ctx,
+		"alpha",
+		"billing",
+		"projects",
+		"link",
+		projID,
+		"--billing-account",
+		accountID,
+	)
+	return err
+}
+
+func command(ctx context.Context, name string, args ...string) ([]string, error) {
```

Comment:
This should be pulled into a separate file like `util.go` so other implementations can share